### PR TITLE
fix popover list height calcs

### LINKED ISSUE #15591: Data selector popover is too small
image

I'm seeing this on Stats with Chrome.

### PR DESCRIPTION
**Pull request by daltojohnso**
fixes #15591 

- revert `Popover` to pre react 16 upgrade 
- move side-effecty logic that depends on rendering to the third arg of `unstable_renderSubtreeIntoContainer`
- remove some config options from ParameterFieldWidget to make it handle being in a sidebar better

### CODE REVIEW

**Comment by daltojohnso** [comment]
Fixing tests...

File: frontend/src/metabase/components/Popover.jsx [hunk 1]
```diff
@@ -246,24 +246,25 @@ export default class Popover extends Component {
...
-          }
-        }
-      }
-
       // finally set the best options
       this._setTetherOptions(tetherOptions, this._best);
     }
+
+    // after tether has done its business, resize the popover body to fit the browser window if so desired
+    if (this.props.sizeToFit) {
```
Comment (howonlee) [inline frontend/src/metabase/components/Popover.jsx:254]: this is going to have 15502 pop back up again, it looks like, tho?

Comment (howonlee) [reply frontend/src/metabase/components/Popover.jsx:254]: I can't repro the shifting ot the popover but I can repro the popover covering the button after exactly 4 clicks again

Comment (daltojohnso) [reply frontend/src/metabase/components/Popover.jsx:254]: oh ya you're right, was able to repro it

Comment (daltojohnso) [reply frontend/src/metabase/components/Popover.jsx:254]: @howonlee check out the change I made here https://github.com/metabase/metabase/pull/15608/commits/6fb13700454dc06830cf8a70b5ca5081d86c3427

this hack seems to fix things for me. anything else broken for you?

**Review by flamber — COMMENTED** [review]
Much improvement! The also seems to fix other weird behavior I've seen for a long time.

I'm having difficulties keeping track of what caused what, so sorry if this is not caused by this PR.

1. Opening a popover multiple times positions the popover in different spot - sometimes out of viewport
Clicking first time
image
Second time
image
2. Some popovers causes it to cover the initiator after a couple of clicks
First click
image
After a couple of clicks
image
3. The binning dimension seems to be small sometimes now - and when it is small, then clicking two more times will make it "disappear" (goes off viewport without scroll indicators)
image
4. Noticed that the popover on native (#15549) now does not trigger a body scroll, so it's not even possible to define default values now.

**Comment by howonlee** [comment]
popover multiple open position mutation is caused by the change of place of the the `sizeToFit` code, I bet, because I fixed it by changing it to be over where it was in the first place. popover covering initiator fix also depended upon that fix. #3 I think is #15445 and I think it was fixed by #15600, but that was after this one was up

**Comment by howonlee** [comment]
but then, reverting the `sizeToFit` change makes the original bug pop up again, like carpet bubbles...

**Comment by daltojohnso** [comment]
with a tiny tweak to old `Popover` code it seems to work much better than new `Popover` + portals code. this is unfortunate, but we can address it in later releases. presumably the refactor introduced some bad logic that just doesn't work well with tether doing things and us doing things to tether when `sizeToFit` is set.

@flamber this might re-introduce anything portals happened to fix, if it fixed anything at all, so mind taking another look around the app?

I made one additional tweak to try to make the default parameter value widget work better in the sidebar by removing some config options to let it render above the widget.

popovers in the notebook still move around a bit when opening up, but I'm not sure if that is fixable.

**Review by flamber — COMMENTED** [review]
Tested commit `24307a1`
It works almost similar to the current 38.4 style - with some of the quirks that has been around forever.
And the Native default filter is much better - not perfect, but fully workable.

But still seeing some problems compared to 38.4 - all really annoying, but getting to a point where they are almost non-blocking.

1. Block-ish: Click the filter will sometimes make the popover go outside of viewport. Clicking again will make it show normally. Can't find the consistency - seems like a race-condition with how the popover options gets populated.
image
2. Annoying: It now creates the double-binning and sometimes shows a very small popover
image
3. Annoying: Sometimes clicking on group by multiple times will make popover position on top
image

**Comment by howonlee** [comment]
can repro the nondeterministic move-the-popover thing. previously it was deterministic after one click

**Comment by daltojohnso** [comment]
@flamber I can def repro a lot of those things you mention. I'm afraid I don't know how fixable they are. Trying a few shots in the dark. Seems like the main issue is a user double clicking and getting in between two render cycles which are needed because after tether positions the els we go in and screw with it to get it to fit into the browser window and then tether needs to go in and reposition things again.

**Comment by daltojohnso** [comment]
Can you try out https://github.com/metabase/metabase/pull/15608/commits/5f6e4c70c6b1b7f3f0ab7904bf4d2301f51c3edd to see how that feels? I think it'll help prevent users from double clicking and getting in between the double render I described above.

**Comment by daltojohnso** [comment]
I opened https://github.com/metabase/metabase/pull/15632 as an alt approach that doesn't revert to pre-react 16 update code so that we can compare behavior

**Review by flamber — APPROVED** [review]
Tested `5f6e4c7` - not signing off on the code, but this seems to be acceptable.
But it feels less "smooth" than 0.38.4, so I think this should become a priority for 39.x not 40

**Review by ariya — APPROVED** [review]

**Pull request merged**